runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/skills/SkillBonus.java

```diff
@@ -44,4 +44,15 @@ public interface SkillBonus
 	 * @return The skill bonus multiplier.
 	 */
 	float getValue();
+
+
+	/**
+	 * Gets the list of skill bonuses this skill bonus can be stacked with.
+	 *
+	 * @return List of stackable skill bonuses
+	 */
+	default SkillBonus[] getCanBeStackedWith()
```

Comment:
I suppose this is mostly to manage prayer bonuses, since there's a rather complex set of bonuses--some of which can be stacked with others, others which can't--which need to be managed? (Mostly zealot robes as they don't affect the sacred bone burner, consuming bones with the sinister offering spell, cremating shades, or scattering ashes)

I worry this implementation could be error-prone as ideally this logically is defining a BiMap (because if A can be stacked with B, B should also be able to be stacked with A), and this API is very verbose and prone to typos. For instance, for PrayerBonus, this would need to be something like the following: (formatting condensed because it is substantially longer when following the project's style guidelines)

```java
enum PrayerBonus implements SkillBonus {
	LIT_GILDED_ALTAR(/* ... */) {
		@Override getCanBeStackedWith() { return [ZEALOT_ROBES]; }
	},
	ECTOFUNTUS(/* ... */) {
		@Override getCanBeStackedWith() { return [ZEALOT_ROBES]; }
	},
	CHAOS_ALTAR(/* ... */) {
		@Override getCanBeStackedWith() { return [ZEALOT_ROBES]; }
	},
	MORYTANIA_DIARY_3_SHADES(/* ... */),
	BONECRUSHER(/* ... */),
	SINISTER_OFFERING(/* ... */),
	DEMONIC_OFFERING(/* ... */),
	SACRED_BONE_BURNER(/* ... */),
	ZEALOT_ROBES(/* ... */) {
		// NOTE: What happens if we add a new entry above which can be stacked
		//       with zealot robes and forget to update this line??
		@Override getCanBeStackedWith() { return [LIT_GILDED_ALTAR, ECTOFUNTUS, CHAOS_ALTAR]; }
	}
}
```

Hence, I wonder if this would be better implemented as a static instance method returning `BiMap<SkillBonus, SkillBonus>`

Reply:
This is mostly ramble, but this does run into an issue where, let's say down the line some skill has 10 possible bonuses all of which stack; even this solution would require `(n ** 2 + n) / 2` entry listings to achieve. It's possible this is a problem far enough out that it's not enough of a concern to deal with now though; I suspect we're never going to reach rs3 levels of stacked xp bonuses.